test/ammo.mjs

```diff
@@ -0,0 +1,1635 @@
+import { expect } from 'chai';
+
+// Helpers
+
+/**
+ * Assert a number to make sure it is a number and not NaN. Check number to be value if value provided.
+ *
+ * @param {number} num - Number to assert.
+ * @param {number} [val] - Value to expect the number to equal.
+ */
+function assertNum(num, val) {
+    expect(num).to.be.a('number');
+    expect(num).to.not.be.NaN;
+
+    if (val !== undefined) {
+        expect(num).to.equal(val);
+    }
+}
+
+/**
+ * Assert an array to make sure it is an array. Check value against provided callback if any.
+ *
+ * @param {*[]} array - Array to assert.
+ * @param {Function} [assertValue] - Callback function to run for each value in array.
+ */
+function assertArray(array, assertValue) {
+    expect(array).to.be.an('object');
+    assertNum(array.length);
+
+    if (assertValue !== undefined) {
+        array.forEach(assertValue);
+    }
+}
+
+/**
+ * Assert an object to be of a certain Ammo type.
+ *
+ * @param {object} val - The value to assert.
+ * @param {string} type - The Ammo type that the value should be of.
+ */
+function assertAmmoType(val, type) {
+    expect(val).to.be.an('object');
+    expect(val.__type).to.include(type);
+}
+
+/**
+ * Whether an object is of a certain Ammo type.
+ *
+ * @param {object} val - The value to assert.
+ * @param {string} type - The Ammo type that the value should be of.
+ * @returns {boolean} Whether the value is of a certain Ammo type.
+ */
+function isOfAmmoType(val, type) {
+    return val ? val.__type.indexOf(type) !== -1 : false;
+}
+
+/**
+ * Assert a Vector 3 to be a vector 3 as well as to match the provided values if any.
+ *
+ * @param {object} vec - The Vec3 to assert.
+ * @param {number} [x] - The expected X value.
+ * @param {number} [y] - The expected Y value.
+ * @param {number} [z] - The expected Z value.
+ */
+function assertVec3(vec, x, y, z) {
+    assertNum(vec.x, x);
+    assertNum(vec.y, y);
+    assertNum(vec.z, z);
+}
+
+// Stub Ammo
+
+// Note: do not use arrow functions in the Ammo root object, since the engine will
```

Comment:
This comment is not applicable here and can be removed.

Reply:
It's removed!